### ISSUE TITLE
[BUG] On media creation two stomp messages are sent for fits generation causing the second to fail weirdly

### ISSUE THREAD

**Issue by nigelgbanks** [Islandora/islandora-starter-site#19]
> N.B Might not be a problem with configuration, might be a bug in Islandora or islandora_fits, but if so I'd imagine someone would have noticed it?

**What steps does it take to reproduce the issue?**

- Installed the site (all the bells and whistles, fedora etc)
- Log in
- Created an Image Repository Item
- Clicked on the Media tab
- Create a new "Original File" / "Preservation Master File" media entity
- Derivatives created successfully (except fits)

A Fits media entity was created, but it was missing all its data and no `xml` file was written to disk. I looked at the ActiveMQ queue, and it had 2 dequeued events for `islandora-connector-fits` where I would expect one.

Funnily enough, if one manually triggers the "FITS - Generate a Technical metadata derivative" via `admin/content` it works perfectly.

* When does this issue occur?

When creating new media entities.

* What did you expect to happen?

For the Fits to be generated and accessible.

**Which version of Islandora are you using?**

2.4.3 (I was using 8a32258f3a564180fbc69a8a80c2c4c0f785ebd4 of this repository)

**Screenshots:**

**Missing Information:**

![Screenshot 2022-11-07 at 21 19 06](https://user-images.githubusercontent.com/487373/200417652-573302f5-b01b-4867-85df-6570305fdc8e.png)

**Expected:**

![Screenshot 2022-11-07 at 21 20 42](https://user-images.githubusercontent.com/487373/200417937-3073d499-552b-467b-8d3c-84d8087d6e4c.png)

**Comment by rosiel** [comment]
Tried and failed to replicate. 😞  I am on [`8a3225`](https://github.com/Islandora/islandora-starter-site/commit/8a32258f3a564180fbc69a8a80c2c4c0f785ebd4) plus a couple of very minor config changes. I got a functioning FITS from the automatic derivatives. I'm sorry I am not sure what's in my activemq dequeue logs. I tried to expose activemq and [hosed my site](https://github.com/Islandora-Devops/isle-dc/pull/304#discussion_r1015991695).

**Comment by nigelgbanks** [comment]
Might be some difference between my setup and `isle-dc` I'm just using a single docker compose file and for the Docker image I'm just pulling down the starter-site and overwriting `test/rootfs/var/www/drupal/assets/patches/default_settings.txt` to use environment variables to override a few settings, none of which are related to fits (pretty much an exact copy of [this](https://github.com/Islandora-Devops/sandbox/blob/main/docker/sandbox-drupal/rootfs/var/www/drupal/assets/default.settings.php)).

**Comment by rosiel** [comment]
Not intermittent.

I just confirmed that "islandora emitting two stomp messages for one event" happens with `make local` too, so it isn't a problem with the starter site (alone).

I also confirmed that it's not just FITS. On both `make starter` and `make local`, when I create an image (two image derivatives configured), I get four messages to the `islandora-connector-houdini` queue. When I create an audio (one audio derivative), I get two messages to the `islandora-connector-homarus` queue.  And for each of those original files, i am also getting two messages in the `islandora-connector-fits` queue.

`islandora-indexing-fcrepo-content` seems to be getting only one message per new node created (good!).
`islandora-indexing-fcrepo-media` gets two messages per new media created (for a media with no Media Use terms, therefore no derivatives).

I suspect that there is something going on with our `islandora` hooks on media creation/save that is causing the double events.

Also, none of this explains why you're seeing a "broken" FITS with no XML file. The derivatives I have all look fine and useable.

**Comment by nigelgbanks** [comment]
It's working today... no idea why I've not changed any configs, I did run the `drush cron` maybe that's required for some reason, but that makes no sense.

**Comment by nigelgbanks** [comment]
I'm going to close this as can't reproduce.

**Comment by Th0m4sR** [comment]
Hi, I faced the same issue some months ago and it seems in previous versions, the filehash.settings.yml used a field called algos to define the hashing algorithms.
The according Drupal module (filehash) is trying to load a field called algorithms in newer versions (3.2.3 in my case), but it seems to have been a field called algos earlier (although I am not sure in which versions).
In the filehash.settings.yml, I renamed this parameter (from "algos" to "algorithms"), imported the configuration afterwards and this seems to have fixed it for me.

I guess this comes from newer versions of the filehash module that implements the function for enabled algorithms like this:

```PHP
  public function getEnabledAlgorithms(): array {
    $config = $this->configFactory->get('filehash.settings')->get('algorithms');
    if (!\is_array($config)) {
      return [];
    }
    return array_intersect_key(static::getAlgorithms(), array_filter($config));
  }
```

No need to reopen the issue but I just wanted to comment this here in case other users face the same error.
